base/traits.hpp

```diff
@@ -59,6 +59,11 @@ using internal::all_different_v;
 template<template<typename...> typename T, typename U>
 inline constexpr bool is_instance_of_v = internal::is_instance_of<T, U>::value;
 
+// Note the argument inversion so that we can use instance_of<T> in a requires
```

Comment:
I don't quite understand this inversion.  I would argue that:
```
requires instance_of<Quantity, T>
```
would read better than:
```
requires instance_of<T, Quantity>
```
If there is a technical reason for the inversion, it should be explained here (I couldn't find one, see https://godbolt.org/z/fYbndKd1c) and we should also invert the `is_instance_of` trait to minimize confusion.

Reply:
In `requires` clauses, you want to be able to write
`{ expression } -> instance_of<Quantity>`

Compare the inversion between https://en.cppreference.com/w/cpp/types/is_base_of and https://en.cppreference.com/w/cpp/concepts/derived_from.